vclib/src/commonMain/kotlin/at/asitplus/wallet/lib/iso/IssuerSignedItemSerializer.kt

```diff
@@ -0,0 +1,107 @@
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.CompositeEncoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+
+object IssuerSignedItemSerializer : KSerializer<IssuerSignedItem> {
+
+    override val descriptor: SerialDescriptor = buildClassSerialDescriptor("IssuerSignedItem") {
+        element("digestID", Long.serializer().descriptor)
+        element("random", ByteArraySerializer().descriptor)
+        element("elementIdentifier", String.serializer().descriptor)
+        element("elementValue", String.serializer().descriptor)
+    }
+
+    override fun serialize(encoder: Encoder, value: IssuerSignedItem) {
+        encoder.encodeStructure(descriptor) {
+            encodeLongElement(descriptor, 0, value.digestId.toLong())
+            encodeSerializableElement(descriptor, 1, ByteArraySerializer(), value.random)
+            encodeStringElement(descriptor, 2, value.elementIdentifier)
+            encodeAnything(value, 3)
+        }
+    }
+
+    private fun CompositeEncoder.encodeAnything(value: IssuerSignedItem, index: Int) {
+        val descriptor = buildClassSerialDescriptor("IssuerSignedItem") {
+            element("digestID", Long.serializer().descriptor)
+            element("random", ByteArraySerializer().descriptor)
+            element("elementIdentifier", String.serializer().descriptor)
+            element("elementValue", buildElementValueSerializer(value.elementValue).descriptor)
+        }
+
+        when (val it = value.elementValue) {
+            is String -> encodeStringElement(descriptor, index, it)
+            is Int -> encodeIntElement(descriptor, index, it)
+            // TODO write tag 1004
+            is LocalDate -> encodeSerializableElement(descriptor, index, LocalDate.serializer(), it)
+            // TODO write tag 1004
+            is Instant -> encodeSerializableElement(descriptor, index, InstantStringSerializer(), it)
+            is Boolean -> encodeBooleanElement(descriptor, index, it)
+            is ByteArray -> encodeSerializableElement(descriptor, index, ByteArraySerializer(), it)
+            else -> Cbor.encode(descriptor, index, this, it)
+        }
+    }
+
+    private inline fun <reified T> buildElementValueSerializer(element: T) = when (element) {
+        is String -> String.serializer()
+        is Int -> Int.serializer()
+        is LocalDate -> LocalDate.serializer()
+        is Instant -> InstantStringSerializer()
+        is Boolean -> Boolean.serializer()
+        is ByteArray -> ByteArraySerializer()
+        is Any -> Cbor.lookupSerializer(element) ?: error("descriptor not found for $element")
+        else -> error("descriptor not found for $element")
```

Comment:
I trust you, but now's the time for clarification: Why `Any` and `else` as separate cases? I do suspect `Any?` is not valid, because the data needs to be present, right?

Reply:
`element` is non-nullable, so it can't be `Any?`. But the compiler says it still may be `Unit`, so we need the `else` branch.